core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java

```diff
@@ -120,24 +121,19 @@ private PlanNode removeRedundantTableScanPredicate(
 
         TupleDomain<ColumnHandle> unenforcedDomain;
         if (predicateDomain.getDomains().isPresent()) {
-            Map<ColumnHandle, Domain> predicateColumnDomains = predicateDomain.getDomains().get();
-
             // table scans with none domain should be converted to ValuesNode
             checkState(node.getEnforcedConstraint().getDomains().isPresent());
             Map<ColumnHandle, Domain> enforcedColumnDomains = node.getEnforcedConstraint().getDomains().get();
 
-            ImmutableMap.Builder<ColumnHandle, Domain> unenforcedColumnDomains = ImmutableMap.builder();
-            for (Map.Entry<ColumnHandle, Domain> entry : predicateColumnDomains.entrySet()) {
-                ColumnHandle columnHandle = entry.getKey();
-                Domain predicateColumnDomain = entry.getValue();
-                Domain enforcedColumnDomain = enforcedColumnDomains.getOrDefault(columnHandle, Domain.all(predicateColumnDomain.getType()));
+            unenforcedDomain = predicateDomain.transformDomains((columnHandle, predicateColumnDomain) -> {
+                Type type = predicateColumnDomain.getType();
+                Domain enforcedColumnDomain = Optional.ofNullable(enforcedColumnDomains.get(columnHandle)).orElseGet(() -> Domain.all(type));
                 predicateColumnDomain = predicateColumnDomain.intersect(enforcedColumnDomain);
                 if (!predicateColumnDomain.contains(enforcedColumnDomain)) {
-                    unenforcedColumnDomains.put(columnHandle, predicateColumnDomain);
+                    return predicateColumnDomain.intersect(enforcedColumnDomain);
```

Comment:
it's already intersected at line 131

Reply:
my bad, was supposed to be removed, thanks